### PR TITLE
Fix update from default branch not being in merge state

### PR DESCRIPTION
**Pull request by tidy-dev**
## Description

Fix regression that broke update from default branch.

Thanks again @cheshire137!

### Screenshots
https://user-images.githubusercontent.com/75402236/121274829-54038380-c899-11eb-97f4-c709d0e0e927.mov

## Release notes

Notes: [Fixed] Update from default branch works as expected now.

### CODE REVIEW

File: app/src/lib/stores/app-store.ts [hunk 1]
```diff
@@ -4522,7 +4522,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
       opState === null ||
       opState.operationDetail.kind !== MultiCommitOperationKind.Merge
     ) {
-      log.error('[mergeBranch] - Not in merge oerpation state')
+      log.error('[mergeBranch] - Not in merge operation state')
```
Comment (cheshire137) [inline app/src/lib/stores/app-store.ts:4531]: I had grown rathoer fond of this typo. 😢😄

**Review by cheshire137 — COMMENTED** [review]
Thank you for the fix!

File: app/src/ui/dispatcher/dispatcher.ts [hunk 2]
```diff
@@ -970,6 +970,38 @@ export class Dispatcher {
...
+    const {
+      multiCommitOperationState,
+      branchesState: { tip },
+    } = this.repositoryStateManager.get(repository)
+
+    if (
+      multiCommitOperationState === null ||
+      multiCommitOperationState.operationDetail.kind !==
+        MultiCommitOperationKind.Merge
+    ) {
```
Comment (sergiou87) [inline app/src/ui/dispatcher/dispatcher.ts]: Feels weird having this kind of check here knowing `mergeBranch` is also used in other merge entry points to do the actual merge (whereas the multi commit operation was initialized somewhere else). I think I'd rather have a new method in the dispatcher for this that `updateBranch` in app.tsx would call, but this is ok.

In any case, I think it'd be good adding a comment right above this `if` to explain why sometimes there is an ongoing Merge operation, and sometimes there isn't 🤔

**Review by sergiou87 — COMMENTED** [review]
LGTM! Only have a comment/suggestion.

File: app/src/ui/dispatcher/dispatcher.ts (hunk 2 above)
Comment (tidy-dev) [reply app/src/ui/dispatcher/dispatcher.ts]: Yeah. I like that better.. this actually broke in 2 places. updateBranch and in the compare mode.

**Review by sergiou87 — DISMISSED** [review]
Thanks for those changes, looks much better now! 😍

**Review by sergiou87 — APPROVED** [review]

**Pull request merged**